### PR TITLE
Fix theming update of shader and code editors

### PR DESCRIPTION
**Pull request by Chaosus**
and for the few status buttons inside `CodeTextEditor`, and warnings colors.

### CODE REVIEW

File: editor/plugins/script_text_editor.cpp
Comment (akien-mga) [inline editor/plugins/script_text_editor.cpp:496]: I know you just moved this code but since I notice it now, would be good to fix. This is bad `TTR` usage for translation as it includes only part of a sentence, so many languages wouldn't be able to provide a valid translation.

This should be something like this:
```suggestion
		// TRANSLATORS: Script error pointing to a line and column number.
		String error_text = vformat(TTR("Error at (%d, %d):"), errors[0].line, errors[0].column) + " " + errors[0].message;
```

Comment (Chaosus) [reply editor/plugins/script_text_editor.cpp]: Done

**Comment by pycbouh** [comment]
Theming changes seem fine, but we need a review from script editor maintainers here to make sure it still works as expected.

**Review by Paulb23 — DISMISSED** [review]
This seems to have broken the connection gutter, and output is spammed with: 
```
ERROR: Index p_gutter = -1 is out of bounds (gutters.size() = 4).
   at: set_line_gutter_item_color (scene/gui/text_edit.cpp:4711)
```

There's also a crash on exit:
<details>
    <summary>backtrace</summary>

    ```
    ================================================================
    handle_crash: Program crashed with signal 11 
    Engine version: Godot Engine v4.0.alpha.custom_build 
    (c3d28ffa742298d7e00d0a08251ae97870e3ad88)
    Dumping the backtrace. Please include this when reporting the bug on 
    https://github.com/godotengine/godot/issues
    [1] /usr/lib/libc.so.6(+0x3cda0) [0x7ffff77dfda0] (??:0)
    [2] SceneTree::get_edited_scene_root() const (??:0)
    [3] ScriptTextEditor::_update_warnings() (??:0)
    [4] ScriptTextEditor::_notification(int) (??:0)
    [5] ScriptTextEditor::_notificationv(int, bool) (??:0)
    [6] Object::notification(int, bool) (??:0)
    [7] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [8] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [9] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [10] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [11] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [12] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [13] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [14] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [15] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [16] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [17] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [18] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [19] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [20] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [21] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [22] Control::_propagate_theme_changed(Node*, Control*, Window*, bool) (??:0)
    [23] Control::remove_child_notify(Node*) (??:0)
    [24] Node::remove_child(Node*) (??:0)
    [25] Node::_notification(int) (??:0)
    [26] Node::_notificationv(int, bool) (??:0)
    [27] CanvasItem::_notificationv(int, bool) (??:0)
    [28] Control::_notificationv(int, bool) (??:0)
    [29] Container::_notificationv(int, bool) (??:0)
    [30] BoxContainer::_notificationv(int, bool) (??:0)
    [31] VBoxContainer::_notificationv(int, bool) (??:0)
    [32] Object::notification(int, bool) (??:0)
    [33] Object::_predelete() (??:0)
    [34] predelete_handler(Object*) (??:0)
    [35] void memdelete<Node>(Node*) (??:0)
    [36] Node::_notification(int) (??:0)
    [37] Node::_notificationv(int, bool) (??:0)
    [38] CanvasItem::_notificationv(int, bool) (??:0)
    [39] Control::_notificationv(int, bool) (??:0)
    [40] Panel::_notificationv(int, bool) (??:0)
    [41] Object::notification(int, bool) (??:0)
    [42] Object::_predelete() (??:0)
    [43] predelete_handler(Object*) (??:0)
    [44] void memdelete<Node>(Node*) (??:0)
    [45] Node::_notification(int) (??:0)
    [46] Node::_notificationv(int, bool) (??:0)
    [47] CanvasItem::_notificationv(int, bool) (??:0)
    [48] Control::_notificationv(int, bool) (??:0)
    [49] Object::notification(int, bool) (??:0)
    [50] Object::_predelete() (??:0)
    [51] predelete_handler(Object*) (??:0)
    [52] void memdelete<Node>(Node*) (??:0)
    [53] Node::_notification(int) (??:0)
    [54] Node::_notificationv(int, bool) (??:0)
    [55] EditorNode::_notificationv(int, bool) (??:0)
    [56] Object::notification(int, bool) (??:0)
    [57] Object::_predelete() (??:0)
    [58] predelete_handler(Object*) (??:0)
    [59] void memdelete<Node>(Node*) (??:0)
    [60] Node::_notification(int) (??:0)
    [61] Node::_notificationv(int, bool) (??:0)
    [62] Viewport::_notificationv(int, bool) (??:0)
    [63] Window::_notificationv(int, bool) (??:0)
    [64] Object::notification(int, bool) (??:0)
    [65] Object::_predelete() (??:0)
    [66] predelete_handler(Object*) (??:0)
    [67] void memdelete<Window>(Window*) (??:0)
    [68] SceneTree::finalize() (??:0)
    [69] OS_LinuxBSD::run() (??:0)
    [70] /home/paulb23/development/godot/bin/godot.linuxbsd.tools.64.llvm(main+0x1f3) [0x5555575e0c03] (??:0)
    [71] /usr/lib/libc.so.6(__libc_start_main+0xd5) [0x7ffff77cab25] (??:0)
    [72] /home/paulb23/development/godot/bin/godot.linuxbsd.tools.64.llvm(_start+0x2e) [    0x5555575e093e] (??:0)
    -- END OF BACKTRACE --
    ================================================================
    ```
</details>

**Comment by Chaosus** [comment]
@Paulb23 I removed script editor changes for now.

**Comment by Chaosus** [comment]
@Paulb23 Fixed by adding `is_visible_in_tree()` check.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
